src/main/scala/net/bdew/ae2stuff/items/AdvWirelessKit.scala

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) bdew, 2014 - 2015
+ * https://github.com/bdew/ae2stuff
+ *
+ * This mod is distributed under the terms of the Minecraft Mod Public
+ * License 1.0, or MMPL. Please check the contents of the license located in
+ * http://bdew.net/minecraft-mod-public-license/
+ */
+
+package net.bdew.ae2stuff.items
+
+import appeng.api.config.SecurityPermissions
+import appeng.api.exceptions.FailedConnection
+import net.bdew.ae2stuff.grid.Security
+import net.bdew.ae2stuff.machines.wireless.{BlockWireless, TileWireless}
+import net.bdew.ae2stuff.misc.AdvItemLocationStore
+import net.bdew.lib.Misc
+import net.bdew.lib.block.BlockRef
+import net.bdew.lib.items.SimpleItem
+import net.minecraft.entity.EntityLivingBase
+import net.minecraft.entity.player.EntityPlayer
+import net.minecraft.item.ItemStack
+import net.minecraft.world.World
+
+import java.util
+
+object AdvWirelessKit
+    extends SimpleItem("AdvWirelessKit")
+    with AdvItemLocationStore {
+  setMaxStackSize(1)
+
+  def checkSecurity(t1: TileWireless, t2: TileWireless, p: EntityPlayer) = {
+    val pid = Security.getPlayerId(p)
+    Security.playerHasPermission(
+      t1.getNode.getGrid,
+      pid,
+      SecurityPermissions.BUILD
+    ) &&
+    Security.playerHasPermission(
+      t2.getNode.getGrid,
+      pid,
+      SecurityPermissions.BUILD
+    )
+  }
+
+  override def onItemRightClick(
+      stack: ItemStack,
+      world: World,
+      player: EntityPlayer
+  ): ItemStack = {
+    import net.bdew.lib.helpers.ChatHelper._
+    if (!world.isRemote && player.isSneaking) {
+      toggleMode(stack)
+      if (getMode(stack) == 0) {
+        player.addChatMessage(
+          L("ae2stuff.wireless.advtool.queueing.activated").setColor(
+            Color.GREEN
+          )
+        )
+      } else {
+        player.addChatMessage(
+          L("ae2stuff.wireless.advtool.binding.activated").setColor(Color.GREEN)
+        )
+      }
+    }
+    stack
+  }
+
+  override def onItemUse(
+      stack: ItemStack,
+      player: EntityPlayer,
+      world: World,
+      x: Int,
+      y: Int,
+      z: Int,
+      side: Int,
+      xOff: Float,
+      yOff: Float,
+      zOff: Float
+  ): Boolean = {
+    import net.bdew.lib.helpers.ChatHelper._
+    val pos = BlockRef(x, y, z)
+    if (!pos.blockIs(world, BlockWireless)) return false
+    if (!world.isRemote) {
+      if (player.isSneaking) {
+        toggleMode(stack)
+        if (getMode(stack) == 0) {
```

Comment:
Could you explicitly set different values for modes using two different constansts

Reply:
Resolved by d0fa5cc